stripcaged.lua

```diff
@@ -0,0 +1,228 @@
+            end
+
+        end
+        if pre_count ~= count then count_cage = count_cage + 1 end
+    end
+    print(string.format('Dumped %s in %s', plural(count, 'item'),
+        plural(count_cage, 'cage')))
+end
+
+local function cage_dump_list(list)
+    local count_total = {}
+    local empty_cages = 0
+    for _, cage in ipairs(list) do
+        local count = {}
+        for _, ref in ipairs(cage.general_refs) do
+            if df.general_ref_contains_itemst:is_instance(ref) then
+                local classname = df.item_type.attrs[
+                    df.item.find(ref.item_id):getType()].caption
+                count[classname] = (count[classname] or 0) + 1
+            elseif df.general_ref_contains_unitst:is_instance(ref) then
+                local inventory = df.unit.find(ref.unit_id).inventory
+                for _, it in ipairs(inventory) do
+                    local classname = df.item_type.attrs[it.item:getType()].caption
+                    count[classname] = (count[classname] or 0) + 1
+                end
+                -- TODO: vermin ?
+
+                --[[ TODO: Determine how/if to handle a DEBUG flag.
+
+                --Ruby:
+                  else
+                      puts "unhandled ref #{ref.inspect}" if $DEBUG
+                  end
+                ]]
+            end
+        end
+
+        local type = df.item_type.attrs[cage:getType()].caption -- Default case
+        if df.item_cagest:is_instance(cage) then
+            type = 'Cage'
+        elseif df.item_animaltrapst:is_instance(cage) then
+            type = 'Animal trap'
+        end
+
+        -- If count is empty
+        if not next(count) then
+            empty_cages = empty_cages + 1
+        else
+            local sortedlist = {}
+            for classname, n in pairs(count) do
+                sortedlist[#sortedlist + 1] = {classname = classname, count = n}
+            end
+            table.sort(sortedlist, (function(i, j) return i.count < j.count end))
+            print(('%s %d: '):format(type, cage.id))
+            for _, t in ipairs(sortedlist) do
+                print(' ' .. t.count .. ' ' .. t.classname)
+            end
+        end
+        for k, v in pairs(count) do count_total[k] = (count_total[k] or 0) + v end
+    end
+
+    if #list > 1 then
+        print '\nTotal:'
+        local sortedlist = {}
+        for classname, n in pairs(count_total) do
+            sortedlist[#sortedlist + 1] = {classname = classname, count = n}
+        end
+        table.sort(sortedlist, (function(i, j) return i.count < j.count end))
+        for _, t in ipairs(sortedlist) do
+            print(' ' .. t.count .. ' ' .. t.classname)
+        end
+        print('with ' .. plural(empty_cages, 'empty cage'))
+    end
+end
+
+-- handle magic script arguments
+
+local args = {...}
+
+local list
+if args[2] == 'here' then
+    print "NOTE: The 'here' option isn't well tested for v50 and only works with the keyboard cursor."
```

Comment:
now that I've actually implemented getSelectedItem for v50, this will probably work just fine when clicking on an item and doesn't need the warning.